File: Backend/Reactivities.Domain/Models/User/Activity.cs
using Reactivities.Domain.Models.Attendee;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Reactivities.Domain.Models.User;

public class Activity
{
	#region Properties

	public Guid Id { get; set; }

	public string Title { get; set; }

	public DateTime Date { get; set; }

	public string Description { get; set; }

	public string Category { get; set; }

	public string City { get; set; }

	public string Venue { get; set; }

    public bool IsCancelled { get; set; }
    public string ImageName { get; set; }

    #endregion

    #region Relations

    public ICollection<ActivityAttendee> ActivityAttendees { get; set; }

    #endregion
}

File: reactivities-app/src/app/models/activity.ts
import { Profile } from "./profile";


export interface Activity{
    id:string;
    title:string;
    date:string;
    description:string;
    category:string;
    city:string;
    venue:string;
    imageName:string;
    isCancelled:boolean;
    attendees:Profile[];
    hostUserName:string;
    isHost?:boolean;
    isGoing?:boolean;
    host?:Profile;
}

 // I Use it
export interface CreateOrEditActivity{
    id:string;
    title:string;
    date:string;
    description:string;
    category:string;
    city:string;
    venue:string;
}

// Or
export class ActivityFormValues {
  id?: string = undefined;
  title: string = "";
  date: Date | null = null;
  description: string = "";
  category: string = "";
  city: string = "";
  venue: string = "";
  constructor(activity : ActivityFormValues ) {
    this.category = activity.category;
    this.id = activity?.id;
    this.date = activity?.date;
    this.venue = activity?.venue;
    this.description = activity?.description;
    this.city=activity.city;
    this.title= activity.title
  }
}

export interface predicate {
  all?: boolean,
  isHost?: boolean,
  isGoing?: boolean,
  startDate?: '' | null
};